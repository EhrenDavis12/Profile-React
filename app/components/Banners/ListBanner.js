import React from 'react';
import PropTypes from 'prop-types';
import Banner from 'components/Banners/Banner';
import { SectionBreak, Ul } from 'components/SharedStyles/styles';

function ListBanner(props) {
  const List = props.list.map(element => <li key={element}>{element}</li>);
  return (
    <Banner>
      <SectionBreak>
        <h4>{props.title}:</h4>
      </SectionBreak>
      <SectionBreak>
        <Ul>{List}</Ul>
      </SectionBreak>
    </Banner>
  );
}

ListBanner.propTypes = {
  list: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
};

export default ListBanner;
