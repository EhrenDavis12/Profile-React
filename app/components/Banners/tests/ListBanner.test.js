import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import ListBanner from '../ListBanner';

const List = ['name1 ', 'name2'];

describe('<BannerSection />', () => {
  it('should render the page', () => {
    const renderedComponent = shallow(
      <ListBanner title="List Of Names" list={List} />,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
