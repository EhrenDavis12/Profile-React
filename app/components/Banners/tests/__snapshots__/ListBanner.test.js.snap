// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerSection /> should render the page 1`] = `
<BannerSection>
  <styles__SectionBreak>
    <h4>
      List Of Names
      :
    </h4>
  </styles__SectionBreak>
  <styles__SectionBreak>
    <styles__Ul>
      <li
        key="name1 "
      >
        name1 
      </li>
      <li
        key="name2"
      >
        name2
      </li>
    </styles__Ul>
  </styles__SectionBreak>
</BannerSection>
`;
